src/crs.rs

```diff
@@ -1,194 +1,172 @@
-use std::{env, fs::File, io::Write, path::PathBuf};
+use bytesize::ByteSize;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use std::env;
 
-use futures_util::StreamExt;
+use crate::{CRSError, Error};
 
-// TODO(blaine): Use manifest parsing in BB instead of hardcoding these
+// TODO(#175): Use manifest parsing in BB instead of hardcoding these
 const G1_START: usize = 28;
 const G2_START: usize = 28 + (5_040_001 * 64);
 const G2_END: usize = G2_START + 128 - 1;
 
-const BACKEND_IDENTIFIER: &str = "acvm-backend-barretenberg";
-const TRANSCRIPT_NAME: &str = "transcript00.dat";
-const TRANSCRIPT_URL: &str =
-    "http://aztec-ignition.s3.amazonaws.com/MAIN%20IGNITION/monomial/transcript00.dat";
-
-fn transcript_location() -> PathBuf {
-    match env::var("BARRETENBERG_TRANSCRIPT") {
-        Ok(dir) => PathBuf::from(dir),
-        Err(_) => dirs::home_dir()
-            .unwrap()
-            .join(".nargo")
-            .join("backends")
-            .join(BACKEND_IDENTIFIER)
-            .join(TRANSCRIPT_NAME),
-    }
-}
+const TRANSCRIPT_URL_ENV_VAR: &str = "TRANSCRIPT_URL";
+const TRANSCRIPT_URL_FALLBACK: &str =
+    "https://aztec-ignition.s3.amazonaws.com/MAIN%20IGNITION/monomial/transcript00.dat";
 
 #[allow(clippy::upper_case_acronyms)]
-#[derive(Clone, Default)]
+#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
 pub(crate) struct CRS {
     pub(crate) g1_data: Vec<u8>,
     pub(crate) g2_data: Vec<u8>,
-    // num_points: usize,
+    pub(crate) num_points: usize,
 }
 
 impl CRS {
-    pub(crate) fn new(num_points: usize) -> CRS {
+    pub(crate) async fn update(&mut self, num_points: usize) -> Result<(), Error> {
+        // We already have some data, so start at the end of our list
+        let g1_start = G1_START + self.g1_data.len();
         // UltraPlonk requires a CRS equal to circuit size plus one!
         // We need to bump our polynomial degrees by 1 to handle zero knowledge
         let g1_end = G1_START + ((num_points + 1) * 64) - 1;
```

Comment:
Should we add a sanity check that g1_and is greater than g1_start? Just in case `((num_points + 1) * 64) - 1` is smaller than `self.g1_data.len()`?

Reply:
We actually do the num_points check in the composer API, but I'm fine wrapping this in an `if`. Pushing now.

Reply:
Ah I missed that you do the `num_points` check there. Perhaps just note it in the method then? It seems unlikely that anyone else would ever be consuming the CRS API